src/services/logger/formatters.ts

```diff
@@ -0,0 +1,96 @@
+import * as Hapi from 'hapi';
+import * as Cookie from 'cookie';
+import * as JWT from 'jsonwebtoken';
+import { compose } from 'ramda';
+import { RequestErrorEvent, RequestSentEvent, LogLine } from './types';
+import { getConfig } from '../../../config';
+
+
+const { auth: { standard } } = getConfig(process.env.NODE_ENV);
+
+
+const tagOrder: (keyof LogLine)[] = [
+  'event',
+  'method',
+  'path',
+  'statusCode',
+  'origin',
+  'userId',
+  'organisationId',
+  'query',
+  'payload',
+  'response',
+];
+
+const errorOrder: (keyof LogLine)[] = ['errorMessage', 'errorStack'];
+
+const extractCookie = (c: string) => {
+  const { [standard.cookie.name]: token } = Cookie.parse(c);
+  return token;
+};
+
+const decodeToken = (c: string): { userId?: number, organisationId?: number } => {
+  let decoded: { userId?: number, organisationId?: number };
+  try {
+    decoded = <any> JWT.verify(c, standard.jwt.secret, standard.jwt.verifyOptions);
+
+  } catch (error) {
+    decoded = {};
+  }
+
+  return decoded;
+};
+
+const getIds = compose(decodeToken, extractCookie);
+
+const extractOrigin = (headers: Hapi.Util.Dictionary<string>) => {
+  return ['x-forwarded-for', 'origin', 'referer']
+  .reduce((acc, k) => acc ? acc : headers[k], null) || '';
```

Comment:
Is this meant to be a find type function where you look for either 'x-forwarded-for', 'origin' or 'referer' headers in order?

`.find` may be more readable.

Reply:
`['a', 'b', 'c'].find(k => headers[k])` would return `a|b|c` instead of the value of those headers.
`Object.keys